playbooks/setup-nodepool.yml

```diff
@@ -0,0 +1,230 @@
+          - python-virtualenv
+          - ufw
+        update_cache: yes
+
+    # install extra packages for diskimage-builder
+    # The native xenial package which provides vhd-utils (blktap-utils) does not support
+    # the 'convert' command which is required in order to properly prepare VHD images for
+    # the Xen hosts used by Rackspace Public Cloud. We therefore make use of the same PPA
+    # used by openstack-infra which has the modified version available.
+    # https://launchpad.net/~openstack-ci-core/+archive/ubuntu/vhd-util
+    # built from: https://github.com/emonty/vhd-util
+    # deployed by: https://github.com/openstack-infra/puppet-diskimage_builder/blob/339340409823927bb987f0195c6cedfdace05f4a/manifests/init.pp#L26
+
+    - name: Add vhd-util PPA
+      apt_repository:
+        filename: "vhd-util"
+        repo: "ppa:openstack-ci-core/vhd-util"
+        update_cache: yes
+
+    - name: Install vhd-util
+      apt:
+        name: "vhd-util"
+
+    # the default /etc/hosts file results in the name of the instance
+    # resolving to its own private address first, causing zookeeper
+    # to listen on the wrong address, and thus clustering to fail
+    - name: Prepare /etc/hosts for the zookeeper group
+      copy:
+        content: |
+          127.0.0.1 localhost
+          # The following lines are desirable for IPv6 capable hosts
+          ::1 ip6-localhost ip6-loopback
+          fe00::0 ip6-localnet
+          ff00::0 ip6-mcastprefix
+          ff02::1 ip6-allnodes
+          ff02::2 ip6-allrouters
+          ff02::3 ip6-allhosts
+          # zookeeper hosts
+          {% for host in zookeeper_hosts %}
+          {{ hostvars[host].ansible_default_ipv4.address }} {{ host }}
+          {{ hostvars[host].ansible_default_ipv6.address }} {{ host }}
+          {% endfor %}
+        dest: "/etc/hosts"
+
+    - name: Configure firewall to allow ssh
+      ufw:
+        rule: allow
+        name: OpenSSH
+
+    - name: Configure firewall to allow cluster traffic (ipv4)
+      ufw:
+        rule: allow
+        from_ip: "{{ hostvars[item].ansible_default_ipv4.address }}"
+      with_items: "{{ zookeeper_hosts }}"
+
+    - name: Configure firewall to allow cluster traffic (ipv6)
+      ufw:
+        rule: allow
+        from_ip: "{{ hostvars[item].ansible_default_ipv6.address }}"
+      with_items: "{{ zookeeper_hosts }}"
+
+    - name: Enable firewall
+      ufw:
+        state: enabled
+        policy: deny
+        direction: incoming
+
+    - include_role:
+        name: "ansible-zookeeper"
+
+    - include_role:
+        name: "ansible-role-diskimage-builder"
+
+    - include_role:
+        name: "ansible-role-nodepool"
+
+    - name: Create openstack config directory
+      file:
+        path: "/var/lib/nodepool/.config/openstack"
+        owner: "nodepool"
+        group: "nodepool"
+        mode: "0700"
+        state: directory
+
+    - name: Copy clouds.yaml
+      copy:
+        src: "~/.config/openstack/clouds.yaml"
```

Comment:
wrong location

Reply:
Yep, AFAIK writeCloudsCfg() won't put clouds.yaml in ~.